File: src/main/java/com/sbuify/SBUify/domain/Album.java
package com.sbuify.SBUify.domain;

import javax.persistence.Entity;
import java.util.Collection;
import java.util.Date;
import java.util.List;

@Entity
public class Album extends CatalogItem implements Queueable {
    public Date releaseDate;
    public Double duration;
    public Integer numSongs;
    private Artist artist;
    private List<Song> songs;

    @Override
    public Collection<Song> getItems() {
        return null;
    }
}

File: client/src/app/album/album.ts
import { CatalogItem } from "../shared/catalog-item";
import { Image } from "../shared/image";
import { Song } from "../songs/song";
import { Artist } from "../artist/artist";
import { User } from "../user/user";
import { Queueable } from "../play-queue/queueable";
import { Playable } from "../playback/playable";

/**
 * Represents an album.
 */
export class Album extends CatalogItem implements Queueable, Playable {
  constructor(
    public id: number,
    public name: string,
    public createdDate: any,
    public active: boolean,
    public image: Image,
    public owner: User,
    public releaseDate: any,
    public length: number,
    public numSongs: number,
    public mbid: string,
    public artist: Artist,
    public type: string,
    public songs: Array<Song>
  ) {
    super(id, name, createdDate, active, image, owner);
  }
}
